lib/streamlit/elements/alert.py

```diff
@@ -12,86 +12,173 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import cast, TYPE_CHECKING
+from typing import cast, Optional, TYPE_CHECKING
 
 from streamlit.proto.Alert_pb2 import Alert as AlertProto
+from streamlit.errors import StreamlitAPIException
 from .utils import clean_text
+import re
 
 if TYPE_CHECKING:
     from streamlit.delta_generator import DeltaGenerator
     from streamlit.type_util import SupportsStr
 
+MATCH_EMOJI = re.compile(
```

Comment:
Move to string_util or somewhere else to reuse code.

Reply:
Would be even better to use a module for this but Vincent was against it I think.

Reply:
> Move to string_util or somewhere else to reuse code.

Done! Updated on [5a04d32](https://github.com/streamlit/streamlit/pull/4964/commits/5a04d32c18ab0333f594d8be5fffa009404d9694)

> Would be even better to use a module for this but Vincent was against it I think.

Yeah, there's been some conversation going around for that, and for now, folks have decided against using a dependency for this. There's a project in the roadmap to improve emoji support, so hopefully we'll find a more resilient solution soon!